stateless.go

```diff
@@ -252,22 +252,19 @@ func (n *StatelessNode) Insert(key []byte, value []byte, resolver NodeResolverFn
 				return nil
 			}
 
-			newhash := &HashedNode{new(Fr), new(Point)}
-			newhash.commitment.SetBytesTrusted(unresolved)
-			toFr(newhash.hash, newhash.commitment)
-			n.children[nChild] = newhash
+			n.children[nChild] = &HashedNode{unresolved}
 			// fallthrough to hash resolution
 		}
 
 		// If the child is a hash, the node needs to be resolved
 		// before there is an insert into it.
 		if h, ok := n.children[nChild].(*HashedNode); ok {
-			comm := h.Commitment().Bytes()
-			serialized, err := resolver(comm[:])
+			comm := h.commitment
+			serialized, err := resolver(comm)
 			if err != nil {
 				return err
 			}
-			node, err := ParseNode(serialized, n.depth+1, comm[:])
+			node, err := ParseNode(serialized, n.depth+1, comm)
```

Comment:
`ParseStatelessNode` should be used here: if you are using a 'stateless' tree, then resolving a node should yield stateless nodes.

So `ParseStatelessNode` would need to be amended to also handle leaf values (i.e. just have the old behavior). See comment above.